trin-core/src/portalnet/overlay.rs

```diff
@@ -259,58 +259,68 @@ where
+                        Ok(content_key) => Some((content_key, content_value)),
                         Err(err) => {
                             warn!("Unexpected error while decoding overlay content key: {err}");
                             None
                         }
                     },
                 )
-                .flatten()
+                .map(|(key, content_value)| {
+                    // Spawn a task that...
+                    // - Validates accepted content (this step requires a dedicated task since it
+                    // might require non-blocking requests to this/other overlay networks)
+                    // - Checks if validated content should be stored, and stores it if true
+                    // - Propagate all validated content
+                    let validator = Arc::clone(&validator);
+                    let storage = Arc::clone(&storage);
+                    tokio::spawn(async move {
+                        // Validated received content
+                        if let Err(err) = validator
+                            .validate_content(&key, &content_value.to_vec())
+                            .await
+                        {
+                            // Skip storing & propagating content if it's not valid
+                            warn!("Unable to validate received content: {err:?}");
+                            return None;
+                        }
+
+                        // Check if data should be stored, and store if true.
+                        // Ignore error since all validated content is propagated.
+                        let _ = storage
+                            .write()
+                            .store_if_should(&key, &content_value.to_vec());
+
+                        Some((key, content_value))
+                    })
+                })
                 .collect();
             let validated_content = join_all(handles)
                 .await
                 .into_iter()
-                .filter_map(|content| content.ok())
+                // Whether the spawn fails or the content fails validation, we don't want it:
+                .filter_map(|content| content.unwrap_or(None))
                 .collect();
             // Propagate all validated content, whether or not it was stored.
-            Self::propagate_gossip(validated_content, kbuckets, command_tx);
+            Self::_propagate_gossip(validated_content, kbuckets, command_tx);
         });
         Ok(())
     }
 
-    /// Propagate gossip accepted content via OFFER/ACCEPT:
-    fn propagate_gossip(
+    /// Propagate gossip accepted content via OFFER/ACCEPT, return number of peers propagated
+    pub fn propagate_gossip(&self, content: Vec<(TContentKey, ByteList)>) -> usize {
+        let kbuckets = Arc::clone(&self.kbuckets);
+        let command_tx = self.command_tx.clone();
+        Self::_propagate_gossip(content, kbuckets, command_tx)
+    }
+
+    fn _propagate_gossip(
```

Comment:
Not sure if it is idiomatic rust to underscore private methods, my understanding is that underscore is used as a way to ignore something in Rust? Took a quick look into some of the most popular rust crates and I can't find this pattern.